Sources/Datasets/datasets/image/MNIST.swift

```diff
@@ -1,99 +1,108 @@
 import Foundation
+import S5TF
+import TensorFlow
 
-public struct MNISTBatch: S5TFLabeledBatch {
-    public var data: Int
-    public var labels: Int
+public struct MNIST: S5TFDataset {
+    typealias DataLoader = MNISTDataLoader
+    public static var train: MNISTDataLoader {
+        get {
+            return MNISTDataLoader(split: .train)
+        }
+    }
+    public static var test: MNISTDataLoader {
+        get {
+            return MNISTDataLoader(split: .test)
+        }
+    }
+
+    public static var info = mnistInfo
+
+    private init() {}
 }
 
-fileprivate let MNISTInfo = S5TFDatasetInfo(
+fileprivate let mnistInfo = S5TFDatasetInfo(
     name: "mnist",
     version: "0.0.1",
     description: "The MNIST database of handwritten digits. 60000 train examples and 10000 test examples with image and label features.",
-    homepage: "http://yann.lecun.com/exdb/mnist/",
-    citation: """
-    @article{lecun2010mnist,
-      title={MNIST handwritten digit database},
-      author={LeCun, Yann and Cortes, Corinna and Burges, CJ},
-      journal={ATT Labs [Online]. Available: http://yann.lecun.com/exdb/mnist},
-      volume={2},
-      year={2010}
-    }
-    """)
-
-public struct MNISTIterator: S5TFDatasetIterator {
-    typealias _Dataset = MNIST // swiftlint:disable:this type_name
-    private var dataset: MNIST
+    homepage: URL(string: "http://yann.lecun.com/exdb/mnist/")!
+)
 
+public struct MNISTDataLoader: S5TFDataLoader {
     private var index = 0
 
-    public init(dataset: MNIST) {
-        self.dataset = dataset
-    }
-
-    mutating public func next() -> MNISTBatch? {
-        guard index < dataset.data.count else { return nil }
-        let data = dataset.data[index]
-        index += 1
-        return MNISTBatch(data: data[0], labels: data[1])
-    }
-}
+    let split: S5TFSplit
+    let batchSize: Int
+    let data: Tensor<UInt8>
+    let labels: Tensor<Int64>
```

Comment:
Please make these compatible with training. Labels should be `Int32` to work with `softmaxCrossEntropy` and I'm pretty sure data needs to be of type `Float` to work with S4TF layers.